src/tests/test_magic.py

```diff
@@ -196,6 +196,193 @@ def test_persist_bare(ip):
     assert result.error_in_exec
 
 
+def get_table_rows_as_dataframe(ip, table_name):
+    save_df_name = f"df_{table_name}"
+    ip.run_cell(f"results = %sql SELECT * FROM {table_name} LIMIT 1;")
+    ip.run_cell(f"{save_df_name} = results.DataFrame()")
+    return save_df_name
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_abbr_no_override(ip, test_table, expected_result):
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql -P sqlite:// {saved_df_name}")
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_no_override(ip, test_table, expected_result):
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_override(ip, test_table, expected_result):
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    # To test it's referring to new saved dataframe
+    assert out.result == expected_result
```

Comment:
How do we know that the values changed? I think we should add another table to make sure the values are changed.